src/com/google/enterprise/adaptor/opentext/OpentextAdaptor.java

```diff
@@ -675,6 +712,236 @@ GroupPrincipal getGroupPrincipal(Member group) {
+    conn.addRequestProperty(COOKIE, "LLCookie=" + authenticationToken);
+    try {
+      DocumentBuilder builder =
+          this.documentBuilderFactory.newDocumentBuilder();
+      return builder.parse(conn.getInputStream()).getDocumentElement();
+    } catch (ParserConfigurationException | SAXException e) {
+      throw new IOException("Failed to get search results", e);
+    }
+  }
+
+  /* Helper to read the text content of a node, excluding element
+   * children. Some search result elements contain the value
+   * we're interested in as text/pcdata plus one or more Value
+   * child elements containing multilingual variants of the data.
+   */
+  private String getTextContent(
+      Element container, String contentElementName) {
+    NodeList list = container.getElementsByTagName(contentElementName);
+    if (list.getLength() != 1) {
+      log.log(Level.FINE,
+          "Missing/unexpected values for {0}", contentElementName);
+      return "";
+    }
+    StringBuilder stringBuilder = new StringBuilder();
+    NodeList children = list.item(0).getChildNodes();
+    for (int i = 0; i < children.getLength(); i++) {
+      org.w3c.dom.Node child = children.item(i);
+      if (child.getNodeType() == CDATA_SECTION_NODE
+          || child.getNodeType() == TEXT_NODE) {
+        stringBuilder.append(child.getTextContent().trim()).append(" ");
+      }
+    }
+    return stringBuilder.toString().trim();
+  }
+
+  @VisibleForTesting
+  String getLastModifiedQuery() {
+    StringBuilder query = new StringBuilder();
+    query.append("func=search&outputFormat=xml&findSimilar=false")
+        .append("&functionMenu=false&hhTerms=false&hitHightlight=false")
+        .append("&sortByRegion=OTModifyDate&sortDirection=asc")
+        .append("&startAt=1&goFor=100");
+    for (int i = 0; i < this.startPoints.size(); i++) {
+      query.append("&Location_ID").append(i + 1).append("=")
+          .append(this.startPoints.get(i).getNodeId());
+    }
+    if (this.lastModDate == null) {
+      // Search within last day.
+      SimpleDateFormat dateFormat = new SimpleDateFormat("yyyyMMdd");
+      this.lastModDate = dateFormat.format(
+          new Date(System.currentTimeMillis() - ONE_DAY_MILLIS));
+      this.lastModTime = "000000";
+    }
+    query.append("&where1=")
+        .append(
+            escapeParam("[QLREGION \"OTModifyDate\"] = \"{0}\"",
+                this.lastModDate))
+        .append("&boolean2=AND")
+        .append("&where2=")
+        .append(
+            escapeParam("[QLREGION \"OTModifyTime\"] > \"{0}\"",
+                this.lastModTime))
+        .append("&boolean3=OR")
+        .append("&where3=")
+        .append(
+            escapeParam("[QLREGION \"OTModifyDate\"] > \"{0}\"",
+                this.lastModDate));
+    return query.toString();
+  }
+
+  private String escapeParam(String param, String... values) {
+    return paramEscaper.escape(
+        MessageFormat.format(param, (Object[]) values));
```

Comment:
I don't think MessageFormat has any shortcuts for no arguments, and this could be pretty expensive in that case, and worse, probably leads to undesirable semantics around single quotes in the param to be escaped. If you really like this helper, instead of inlining it above, you could check for values.length == 0, or just have 2 overloads, with an without an argument (thus eliminating the cast as well).